### PR TITLE
 COMMERCE-11162 Supplier Role to manage Price Lists and Promotions

### PR DESCRIPTION
**Pull request by brianikim**
Resent from: https://github.com/liferay-commerce/liferay-portal/pull/3871

https://issues.liferay.com/browse/COMMERCE-11162

@luca-pellizzon ,

New PR to handle your commits from COMMERCE-10889.

PriceListModelResourcePermissionLogic handles the custom logic. 

Commit: aba5a0acc45b1167596d604d4af2943a20e0e9d0 checks the supplier accounts assigned to the user and validate whether the user has accountRole `AccountRoleConstants.ROLE_NAME_ACCOUNT_SUPPLIER`.

The regular role, `AccountRoleConstants.ROLE_NAME_SUPPLIER`, will mainly be used to handle portletResourcePermissions  for control panel access and also permissions added in: 54264a87c5e8324b31d8ca37575e0e4d6d032f6c

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:commerce-functional

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:commerce-functional

**Comment by brianikim** [comment]
ci:test:commerce-functional

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:commerce-functional

**Comment by brianikim** [comment]
Test failures not related I believe @luca-pellizzon

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/util/CommerceAccountRoleHelperImpl.java [hunk 1]
```diff
@@ -220,6 +222,22 @@ else if (name.equals(AccountRoleConstants.ROLE_NAME_SUPPLIER)) {
 					new String[] {ActionKeys.ACCESS_IN_CONTROL_PANEL});
 			}
 
+			companyResourceActionIds.put(
+				"com.liferay.commerce.currency",
+				new String[] {
+					CommerceCurrencyActionKeys.MANAGE_COMMERCE_CURRENCIES
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/util/CommerceAccountRoleHelperImpl.java:228]: Probably better to use the local service where it's needed instead of giving the supplier the ability to actually change currencies.
Even better would be having a VIEW permission on currencies, set as guest-default.
Align with @dsitu that faced a similar issue and used the local service

Comment (brianikim) [reply modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/util/CommerceAccountRoleHelperImpl.java:228]: hi @luca-pellizzon , I will identify and adjust accordingly.

Comment (brianikim) [reply modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/util/CommerceAccountRoleHelperImpl.java]: @luca-pellizzon , see commit e283358.

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:commerce-functional

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:commerce-functional

**Comment by brianikim** [comment]
ci:test:commerce-functional

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:commerce

**Comment by brianikim** [comment]
Test failures due to documented QA failure on the CI side: https://github.com/brianchandotcom/liferay-portal/commit/10decbf7f21364f7611fd25c309cd2ed855fabdb. Need this commit to be in the upstream before `:relevant` and `:commerce` can be run again. Let me know if the PR needs any more changes @luca-pellizzon

**Review by luca-pellizzon — APPROVED** [review]

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:commerce-functional

**Comment by brianikim** [comment]
@pietrobocale, could you let me know if the test failure for is related to my PR: CommerceAccountSupplierRole#CanSupplierUserManageCatalogs

I checked in testray and it was seen in other PRs I think.

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:commerce-functional

**Comment by brianikim** [comment]
@luca-pellizzon , see https://github.com/liferay-commerce/liferay-portal/pull/3904/commits/b2b3fb1a8cb0be5e73b4de862792c79332a6efad

Commit for edge case scenario, reindex did not occur for priceLists on updating accountId for commerceCatalogs.

cc/ @pietrobocale, @lorenzolapescara

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:commerce-functional

**Comment by brianikim** [comment]
This test CanSupplierUserManageCatalogs needs to be altered @lorenzolapescara. User must be assigned to Supplier Account and also have accountRole ROLE_NAME_ACCOUNT_SUPPLIER = "Account Supplier" assigned in order to view.

**Comment by pietrobocale** [comment]
ci:test:relevant

**Comment by pietrobocale** [comment]
ci:test:commerce-functional

**Comment by brianikim** [comment]
@luca-pellizzon , see https://github.com/liferay-commerce/liferay-portal/pull/3904/commits/fdcc2ae43b55b0a09bfe38b61147248457a69a04

This is not a new commit. I accidentally removed the commit in my previous update. Basically same commit as one such as in: https://github.com/liferay-commerce/liferay-portal/pull/3912/commits/65e8220fccd4b99ca2cc8b7f54a7c7a461b6f44c . COMMERCE-10891 did not make the alteration based on our new use-case which is why the update was needed.

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:commerce-functional

**Comment by pietrobocale** [comment]
The failures are not related to the PR

**Comment by victoroscuro** [comment]
ci:forward:force
